tools/cdb2sql/cdb2sql.c

```diff
@@ -144,6 +142,182 @@ void cdb2sql_usage(int exit_val)
     exit(exit_val);
 }
 
+
+const char *words[] = {
+  "ALTER", "ANALYZE", 
+  "BEGIN", 
+  "COMMIT",
+  "CREATE", 
+  "DELETE", "DROP", "DRYRUN",
+  "EXEC", "EXPLAIN",
+  "INSERT", 
+  "PUT",
+  "REBUILD",
+  "ROLLBACK",
+  "SELECT", "SELECTV", "SET",
+  "TRUNCATE",
+  "UPDATE",
+  "WITH", NULL, }; // must be terminated by NULL
+
+
+
+// Generator function for word completion.
+char *level_one_generator (const char *text, int state)
+{
+    static int list_index, len;
+    const char *name;
+    if (!state) { //if state is 0 get the length of text
+        list_index = 0;
+        len = strlen (text);
+    }
+    while (name = words[list_index]) {
+        list_index++;
+        if (len == 0 || strncasecmp (name, text, len) == 0) {
+            return strdup (name);
+        }
+    }
+    return ((char *) NULL); // If no names matched, then return NULL.
+}
+
+char *db_generator (int state, const char *sql)
+{
+    static char **db_words;
+    static int list_index, len;
+    const char *name;
+    if (!state) { //if state is 0 get the completions from the db
+        cdb2_hndl_tp *cdb2h_2 = NULL; // use a new db handle
+        if (db_words) {
+            char *wrd;
+            list_index = 0;
+            while ((wrd = db_words[list_index])) {
+                free(wrd);
+                db_words[list_index] = NULL;
+                list_index++;
+            }
+            free(db_words);
+            db_words = NULL;
+        }
+
+        list_index = 0;
+        int rc;
+        if (dbhostname) {
+            rc = cdb2_open(&cdb2h_2, dbname, dbhostname, CDB2_DIRECT_CPU);
+        } else {
+            rc = cdb2_open(&cdb2h_2, dbname, dbtype, 0);
+        }
+        if (rc) {
+            if (debug_trace)
+                fprintf(stderr, "cdb2_open rc %d %s\n", rc, cdb2_errstr(cdb2h));
+            cdb2_close(cdb2h_2);
+            return ((char *) NULL);
+        }
+        rc = cdb2_run_statement(cdb2h_2, sql);
+        if (rc) {
+            fprintf(stderr, "failed to run sql '%s'\n", sql);
```

Comment:
It is okay if the statement can't be auto-completed (for instance, the backend is R6 which does not have completion table). Hence we should silently ignore instead of printing an error here.